src/main/java/io/debezium/connector/vitess/VitessErrorHandler.java

```diff
@@ -48,4 +54,17 @@ protected boolean isRetriable(Throwable throwable) {
         }
         return false;
     }
+
+    private boolean handleConnectionRefusedException() {
+        Integer restarts = connectionRefusedRestarts.get();
+        if (restarts == null) {
+            connectionRefusedRestarts.set(1);
```

Comment:
Also maybe it would be better to have `new AtomicInt(100)` and here just decrement until it hits 100 

```
return restarts.getAndDecrement() <= 100
```

Reply:
Also maybe it would be good to restart the count when exception was handled successfully

Reply:
agree with changes about the decrement, but there's no way to track that the retry was successful, especially from this scope.